### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by kazenaite7**
- Idea fields validation

Extra: exception handling in controller

### CODE REVIEW

File: src/main/java/com/itacademy/goodworks/beans/Idea.java [hunk 1]
```diff
@@ -2,16 +2,38 @@
...
+    @Size(min = 1, max = 77)
     private String organization;
+
+    @Size(min = 1, max = 77)
     private String website;
+
+    @NotNull(message = "optimalParticipatorsAmount cant be null")   //not sure
     private String optimalParticipatorsAmount;
+
     private JobCategory[] category;
```
Comment (grkarolis) [inline src/main/java/com/itacademy/goodworks/beans/Idea.java]: Why you need an array of ``JobCategory`` enums?

File: src/main/java/com/itacademy/goodworks/beans/Idea.java [hunk 2]
```diff
@@ -2,16 +2,38 @@
...
 import org.springframework.data.annotation.Id;
 
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
 public class Idea {
     @Id
     private String id;
+
+    @NotNull(message = "name cant be null")
```
Comment (grkarolis) [inline src/main/java/com/itacademy/goodworks/beans/Idea.java]: For ``Strings`` you should prefer ``@NotBlank`` which trims the string value and checks if the length is greater than zero. This prevents from sending field with spaces or just an empty field and prevents ``null`` values as well.

File: src/main/java/com/itacademy/goodworks/beans/Idea.java (hunk 1 above)
Comment (kazenaite7) [reply src/main/java/com/itacademy/goodworks/beans/Idea.java]: job can have multiple categories

Comment (grkarolis) [reply src/main/java/com/itacademy/goodworks/beans/Idea.java]: Then you should not use an ``array``. I would suggest you using ``List`` instead.

File: src/main/java/com/itacademy/goodworks/beans/JobCategory.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
     private final String name;
 
     JobCategory(String name) {
```
Comment (grkarolis) [inline src/main/java/com/itacademy/goodworks/beans/JobCategory.java]: consider making constructor ``private`` not ``package private``

Comment (kazenaite7) [reply src/main/java/com/itacademy/goodworks/beans/JobCategory.java]: intelij suggests making it private is redudant

File: src/main/java/com/itacademy/goodworks/beans/Idea.java (hunk 1 above)
Comment (kazenaite7) [reply src/main/java/com/itacademy/goodworks/beans/Idea.java]: why?

**Pull request merged**